pkg/harvester/edit/kubevirt.io.virtualmachine/index.vue

```diff
@@ -326,25 +324,40 @@ export default {
     },
 
     restartVM() {
-      if ( this.mode === 'edit') {
-        const cloneDeepNewVM = clone(this.value);
-
-        delete cloneDeepNewVM?.metadata;
-        delete this.cloneVM?.metadata;
+      if (this.mode !== 'edit') {
+        return;
+      }
+      if (!this.value.isRunning) {
+        return;
+      }
+      const cloneDeepNewVM = clone(this.value);
 
-        const oldVM = JSON.parse(JSON.stringify(this.cloneVM));
-        const newVM = JSON.parse(JSON.stringify(cloneDeepNewVM));
+      delete cloneDeepNewVM?.metadata;
+      delete this.cloneVM?.metadata;
+      delete this.cloneVM?.__clone;
 
-        if (!isEqual(oldVM, newVM) && this.isRestartImmediately) {
-          if (this.value.hasAction('restart')) {
-            this.value.doActionGrowl('restart', {});
-          }
+      const oldVM = JSON.parse(JSON.stringify(this.cloneVM));
+      const newVM = JSON.parse(JSON.stringify(cloneDeepNewVM));
 
-          if (this.value.hasAction('start')) {
-            this.value.doActionGrowl('start', {});
-          }
-        }
+      if (isEqual(oldVM, newVM)) {
```

Comment:
You'll find out `oldVM` is not constantly equivalent to `newVM` since `oldVM.spec.template.spec.domain.resources.limits.cpu` is `String`, but the new one is `Number`.
So I convert the new one's cpu to `String` in the `harvester-vm` mixin. @WuJun2016